app/(teacher)/scan/scanned.tsx

```diff
@@ -1,28 +1,144 @@
 import { ThemedText } from "@/components/ThemedText";
 import axios from "axios";
-import { useLocalSearchParams } from "expo-router";
+import { router, useLocalSearchParams } from "expo-router";
 import { useEffect, useState } from "react";
-import { View, Button } from "react-native";
+import { View, Button, StyleSheet, Text } from "react-native";
 
-export default function scanned(){
+interface Person{
+  id: number;
+  first_name: string;
+  last_name: string;
+  email: string;
+}
+
+interface Permission {
+  id: number;
+  parent: string;
+  state: string;
+  start_date: string;
+  end_date: string;
+  qr_code: string;
+}
+
+interface Child{
+  id: number;
+  name: string;
+  surname: string;
+  birth_date: string;
+  classroom_id: number;
+}
+
+export default function Scanned(){
     const { id } = useLocalSearchParams();
     const [data, setData] = useState('');
```

Comment:
Tu wrzucasz data id a nie data. Jak to jest tylko do sprawdzania czy dane przyszły to daj wartość boolean. A jak nie to zmien nazwe na dataId na przykład.